runtime/testy/runtime.py

```diff
@@ -47,11 +47,14 @@ def runtime():
       print("Starting studentCode attempt: %s" % (restartCount,))
       spawnProcess(PROCESS_NAMES.STUDENT_CODE, runStudentCode)
       while True:
-        newBadThing = badThingsQueue.get(block=True)
-        print(RUNTIME_CONFIG.DEBUG_DELIMITER_STRING.value)
-        print(newBadThing)
-        if newBadThing.event in restartEvents:
-          break
+        try:
```

Comment:
I found this a bit strange also. When running on the Beaglebone the main studentCode function would run two iterations and then hang forever. After trying to locate where the error was occurring it was happening after the second call to BadThingsQueue.put() where for some reason the object wasn't added. For some reason setting a delay when BadThingsQueue.get() was called fixed this issue. It may be some type of locking issue, though I'm not completely sure why it would only occur with only 1 CPU running.